_posts/2025-10-29-opensearch-new-ppl-capabilities.md

```diff
@@ -0,0 +1,185 @@
+  meta_description: OpenSearch's Piped Processing Language (PPL) evolves significantly with new and enhanced capabilities that reshape how you handle log analytics and observability workflows. This comprehensive update streamlines how you troubleshoot applications, monitor system performance, and analyze security events, providing essential tools to extract meaningful insights from your observability data. Through enhanced features and refined functionality, teams can navigate complex log analysis with greater precision and clarity.
+  has_math: false
+  has_science_table: false
+---
+
+OpenSearch's Piped Processing Language (PPL) evolves significantly with new and enhanced capabilities that reshape how you handle log analytics and observability workflows. This comprehensive update streamlines how you troubleshoot applications, monitor system performance, and analyze security events, providing essential tools to extract meaningful insights from your observability data. Through enhanced features and refined functionality, teams can navigate complex log analysis with greater precision and clarity.
+
+## What's new in OpenSearch PPL?
+Let's explore the new PPL commands and functions through practical examples of common log analytics use cases. These examples demonstrate how PPL enhanced capabilities can help you analyze logs more effectively, from combining multiple data sources to processing unstructured log data and performing time-series analysis. We'll also cover significant performance improvements in this release, including the integration with Apache Calcite as the query engine.
+
+### 1. New commands and functions
+The OpenSearch 3.3 (https://opensearch.org/blog/explore-opensearch-3-3/) release marks a substantial expansion of PPL functionality with the introduction of 9 new commands, 7 evaluation functions, and 8 statistical functions. The syntax of existing commands has also been refined for improved usability, creating a more intuitive experience for users across various analytical scenarios. Below are scenarios where new commands and functions can help you analyze your data:
+
+#### Combine datasets within single queries ####
+This release enhances PPL data manipulation capabilities with new commands for flexible data combination and field operations. The `append` command combines results from multiple queries into a unified dataset, enabling users to merge data from different sources or time ranges within a single operation. The `join` command combines two datasets together, the left side could be an index or results from a piped commands, the right side could be either an index or a `subsearch`.
+
+Example for using `join`
+
+```
+# Combines web log data with geographical IP data. 
+# Which allowing you to see which countries generate the most traffic.
+PPL> source = web_logs
+  | join type=inner client_ip [source=ip_geodata]
+  | stats count() as total_requests by country
```

Comment:
I don't know if it's in scope for this to be beginner-friendly, but I think this is a complicated query to start with.

Including some simpler starting points per the previous comments might make this easier to digest.

Reply:
I don't think the blog for a beginner. we are trying to highlight some of the new capabilities driven by log analytics real use cases. 
We have enough easy straight forward examples highlighted in documentation and some previous blogs